zerver/views/streams.py

```diff
@@ -518,27 +532,86 @@ def remove_subscriptions_backend(
     return json_success(request, data=result)
 
 
-def you_were_just_subscribed_message(
-    acting_user: UserProfile, recipient_user: UserProfile, stream_names: Set[str]
+def make_message_for_subscription_change(
+    acting_user: UserProfile,
+    recipient_user: UserProfile,
+    stream_names: Set[str],
+    *,
+    add_subscription: bool = False,
+    remove_subscription: bool = False,
```

Comment:
I think introducing `you_were_just_unsubscribed_message` function would be a better choice -- We can avoid the bunch of if..else statements. The `add_subscription` and `remove_subscription` is mutually exclusive, right? So, we don't need to use two arguments here in any case.

Reply:
Yes, you are right. I have refactored this as suggested.